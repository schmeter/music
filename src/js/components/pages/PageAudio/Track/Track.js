import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Mp3File from '../../../Mp3File';
import IconPlay from '../IconPlay';

class Track extends React.Component {
    render() {
        const { file, className, activeTrack } = this.props;
        const playing = activeTrack ? file.path === activeTrack.path : false;

        return (
            <div
                className={classNames(
                    className,
                    'track',
                    { 'track-playing': playing },
                )}
            >
                <span className="table-cell index">
                    {playing ? <IconPlay /> : file.tag.track}
                </span>
                <span className="table-cell">
                    <Mp3File file={file} />
                </span>
            </div>
        );
    }
}

Track.propTypes = {
    className: PropTypes.string,
    file: PropTypes.object.isRequired,
    activeTrack: PropTypes.object,
};

export default Track;
