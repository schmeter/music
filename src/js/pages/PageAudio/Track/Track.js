import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Mp3File from '../../../components/Mp3File';
import IconPlay from '../IconPlay';

const Track = ({
    track,
    className,
    activeTrack,
}) => {
    const playing = track.path === activeTrack?.path;

    return (
        <div
            className={classNames(
                className,
                'track',
                playing && 'track-playing',
            )}
        >
            <span className="table-cell index">
                {playing ? <IconPlay /> : track.tag.track}
            </span>
            <span className="table-cell">
                <Mp3File file={track} />
            </span>
        </div>
    );
};

Track.propTypes = {
    track: PropTypes.object.isRequired,
    className: PropTypes.string,
    activeTrack: PropTypes.object,
};

export default Track;
