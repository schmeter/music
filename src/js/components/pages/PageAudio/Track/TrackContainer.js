import { connect } from 'react-redux';

import Track from './Track';

const mapStateToProps = (state) => ({
    activeTrack: state.audio.activeTrack
});

export default connect(mapStateToProps, null)(Track);
