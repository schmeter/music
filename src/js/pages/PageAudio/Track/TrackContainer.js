import { connect } from 'react-redux';

import Track from './Track';
import { getActiveTrack } from '../../../state/selectors/audio';

const mapStateToProps = state => ({
    activeTrack: getActiveTrack(state),
});

export default connect(mapStateToProps, null)(Track);
