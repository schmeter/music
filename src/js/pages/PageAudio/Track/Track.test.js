import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Track from './Track';

// eslint-disable-next-line react/display-name
jest.mock('../../../components/Mp3File', () => () => <div />);
// eslint-disable-next-line react/display-name
jest.mock('../IconPlay', () => () => <div />);

describe('Track', () => {
  it('renders correctly without active track', () => {
    const props = {
      track: {
        path: 'test',
        tag: {
          track: 1,
        },
      },
    };
    const component = shallow(<Track {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with active track', () => {
    const props = {
      track: {
        path: 'test',
        tag: {
          track: 1,
        },
      },
      activeTrack: {
        path: 'test',
      },
    };
    const component = shallow(<Track {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
