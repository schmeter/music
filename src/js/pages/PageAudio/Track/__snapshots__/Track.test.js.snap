// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track renders correctly with active track 1`] = `
<div
  className="track track-playing"
>
  <span
    className="table-cell index"
  >
    <Component />
  </span>
  <span
    className="table-cell"
  >
    <Component
      file={
        Object {
          "path": "test",
          "tag": Object {
            "track": 1,
          },
        }
      }
    />
  </span>
</div>
`;

exports[`Track renders correctly without active track 1`] = `
<div
  className="track"
>
  <span
    className="table-cell index"
  >
    1
  </span>
  <span
    className="table-cell"
  >
    <Component
      file={
        Object {
          "path": "test",
          "tag": Object {
            "track": 1,
          },
        }
      }
    />
  </span>
</div>
`;
